core/iwasm/libraries/libc-wasi/sandboxed-system-primitives/src/posix.c

```diff
@@ -459,8 +459,25 @@ fd_determine_type_rights(os_file_handle fd, __wasi_filetype_t *type,
                          __wasi_rights_t *rights_inheriting)
 {
     struct __wasi_filestat_t buf;
-    __wasi_errno_t error = os_fstat(fd, &buf);
+    __wasi_errno_t error;
+
+    if (fd == STDIN_FILENO) {
```

Comment:
CI reported compilation error in windows platform:
https://github.com/bytecodealliance/wasm-micro-runtime/actions/runs/10283543347/job/28464839647?pr=3694
```
D:\a\wasm-micro-runtime\wasm-micro-runtime\core\iwasm\libraries\libc-wasi\sandboxed-system-primitives\src\posix.c(464,15):
 error C2065: 'STDIN_FILENO': undeclared identifier
 [D:\a\wasm-micro-runtime\wasm-micro-runtime\product-mini\platforms\windows\build\libiwasm.vcxproj]
```

Maybe we can use code like in wasm_runtime_common.c:
```C
    os_file_handle stdin_file_handle = os_convert_stdin_handle(stdinfd);
    if (fd == stdin_file_handle) {
        ..
    }
    os_file_handle stdout_file_handle = os_convert_stdout_handle(stdoutfd);
    if (fd == stdout_file_handle) {
        ..
    }
    os_file_handle stderr_file_handle = os_convert_stderr_handle(stdoutfd);
    if (fd == stderr_file_handle) {
        ..
    }
```